src/model/consent/consent.ts

```diff
@@ -42,45 +42,62 @@
  * of 'createdAt' field may need to be tested in the future.
  */
 
-import { NotFoundError, RevokedConsentModificationError } from '../errors'
-import Knex from 'knex'
+import { NotFoundError } from '../errors'
+import Knex from 'knex';
 
 /*
  * Interface for Consent resource type
  */
 export interface Consent {
```

Comment:
So the `Consent` model was built around the idea of asking the auth-service for a challenge, storing a partial active consent and then later it would have a credential later be verified and updated on the `Consent` object.

Our API has changed to where the DFSP derives the challenge, PISP user signs and then the DFSP sends the `Consent` creation request alongside a credential to be verified.

So my thinking is 
- that the `credential*` fields should no longer be nullable since they are available on creation of the `Consent` object
- all credentials in `Consent` objects are verified since the `registerConsent` flow verifies before storing, bringing into question if `credentialStatus` is still needed
- `Consent` objects never need to be updated in the traditional sense. The only "updating" that happens is the revoking of the `Consent`. So I've removed all update logic that fell into the old logic of "create `Consent` object now, verify/update credential later"

The DB `schema` ditates a lot and I could use a second opinion @lewisdaly , as well as on any `TODO` you see as well.

Other than those refactoring. I've added two fields that we should need later in for assertion and remove fields that I don't think necessary anymore.